sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -123,9 +124,11 @@ public static Read read() {
     return new AutoValue_RabbitMqIO_Read.Builder()
         .setQueueDeclare(false)
         .setExchangeDeclare(false)
+        // this is not a great policy, this should be set to something more sensible
+        .setRecordIdPolicy(RecordIdPolicy.alwaysUnique())
```

Comment:
I'm actually tempted to get rid of this policy and default to one of the following:
1. sha256
2. always fail immediately and loudly

Deduping messages is really mandatory for rabbit, since CheckpointMark will often fail to serialize state with unacked messages and they *will* be reprocessed.

Reply:
😬 hm, I'm not sure exactly what is best here.  Any thoughts on this from the Beam folks as far as what Beam consumers might expect?  I'd lean towards always failing with an actionable message, personally.

Reply:
I'll ask. I'm not sure what I think is appropriate. I think the best thing to do would be default to a chain-of-command with this priority:
- message id
- timestamp + sha256 (if message id property not set)
- explode loudly (if message id and timestamp properties not set)

Reply:
Current plan is to address this via documentation (which I've written in class-level documentation).